workaround_buggy_warnings_as_errors_return_value.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+workaround_buggy_warnings_as_errors_return_value() {
+  _TEMP_FILE=$(mktemp mix_compile_stderr.XXXXXXXX)
+
+  # prevents warnings in dependencies from being confused with warnings in the current project
+  mix deps.compile
+
+  # the following pair of lines emulate the behaviour of tee for stderr
+  mix compile --force 2> "$_TEMP_FILE"
```

Comment:
For colored output:
```suggestion
  script -qc "mix compile --force" "$_TEMP_FILE"
```
Beware that it requires the util-linux package.

References:
- [linux - How to write stdout to file with colors? - Stack Overflow](https://stackoverflow.com/questions/27397865/how-to-write-stdout-to-file-with-colors)
- [How can I install script command in alpine · Issue #160 · gliderlabs/docker-alpine](https://github.com/gliderlabs/docker-alpine/issues/160)